WikiWatershed/model-my-watershed
# Only run ARA Analysis if within ARA Perimeter

## Overview

Adds a hand-drawn perimeter of the ARA raster to check the area of interest against when deciding if it should be queried or not. If it is not within the perimeter, `null`s are returned. The shape was drawn by hand as it was the most expedient way to do it:

![image](https://user-images.githubusercontent.com/1430060/71526894-e3d09000-28a6-11ea-8bf3-c5f929b72c32.png)

The perimeter is conservative, which is to say that it is smaller than the actual boundary. This is done in the hopes of preempting any edge cases that may query non-existing tiles:

![image](https://user-images.githubusercontent.com/1430060/71527205-5db54900-28a8-11ea-8773-1f8504f54b7e.png)

This is done without precision. If there is a better way of generating this perimeter from a raster, I'd be up for trying that.

Connects #3209 

### Demo

![image](https://user-images.githubusercontent.com/1430060/71525819-838b1f80-28a1-11ea-89ab-4fb152730f5c.png)

## Testing Instructions

* Check out this branch
* Debug Celery `$ ./scripts/debugcelery.sh`
* Go to a place in the map other than the Mid Atlantic or North East, select an area of interest
    - [x] Ensure that all analyses complete successfully, including Land
* Select an area of interest within the Mid Atlantic or North East region
    - [x] Ensure that all analyses complete successfully, including Land
